### PR TITLE
JDK-8314211: Add NativeLibraryUnload event

### PR DESCRIPTION
**Pull request by MBaesken**
[JDK-8313251](https://bugs.openjdk.org/browse/JDK-8313251) introduced a a NativeLibraryLoad event that provides us more detail about shared lib/dll loads.
There should be a similar event for unload operations of shared libs.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8314211](https://bugs.openjdk.org/browse/JDK-8314211): Add NativeLibraryUnload event (**Bug** - P3)

### Reviewers
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to [ce607b38](https://git.openjdk.org/jdk/pull/15272/files/ce607b3885c5d1d86f3441c687d7fee5aa2ce854)
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15272/head:pull/15272` \
`$ git checkout pull/15272`

Update a local copy of the PR: \
`$ git checkout pull/15272` \
`$ git pull https://git.openjdk.org/jdk.git pull/15272/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15272`

View PR using the GUI difftool: \
`$ git pr show -t 15272`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15272.diff">https://git.openjdk.org/jdk/pull/15272.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15272#issuecomment-1677520397)

### CODE REVIEW

**Review by tstuefe — APPROVED** [review]
LGTM. 

While looking at this, I saw that many dlopens in hotspot and jdk coding are not tracked, is that deliberate?

Cheers, Thomas

**Comment by MBaesken** [comment]
> LGTM.
> 
> While looking at this, I saw that many dlopens in hotspot and jdk coding are not tracked, is that deliberate?
> 

Hi Thomas, thanks for the review !

Regarding untracked dlopen in HS codebase, I added a few  with
https://bugs.openjdk.org/browse/JDK-8313616
8313616: support loading library members on AIX in os::dll_load

I also had a question/discussion with David recently about untracked dlopen calls in HS and the outcome was that it was just seen not that important to track those .

Regarding untracked dlopen in JDK C codebase I have this one open for review 
https://bugs.openjdk.org/browse/JDK-8313764
8313764: Offer JVM HS functionality to shared lib load operations done by the JDK codebase

File: src/hotspot/os/windows/os_windows.cpp [hunk 1]
```diff
@@ -1254,13 +1254,36 @@ void  os::dll_unload(void *lib) {
...
+    size_t tl = os::lasterror(buf, sizeof(buf));
     Events::log_dll_message(nullptr, "Attempt to unload dll \"%s\" [" INTPTR_FORMAT "] failed (error code %d)", name, p2i(lib), errcode);
     log_info(os)("Attempt to unload dll \"%s\" [" INTPTR_FORMAT "] failed (error code %d)", name, p2i(lib), errcode);
+#if INCLUDE_JFR
+    event.set_success(false);
+    if (tl != 0) {
+      event.set_errorMessage(buf);
+    } else {
+      os::snprintf(buf, sizeof(buf), "Attempt to unload dll failed (error code %d)", (int) errcode);
+      event.set_errorMessage(buf);
```
Comment (TheRealMDoerr) [inline src/hotspot/os/windows/os_windows.cpp:1283]: Please move replicated line after if-then-else block. Otherwise, good.

Comment (MBaesken) [reply src/hotspot/os/windows/os_windows.cpp]: Hi Martin, I adjusted the coding you mentioned .

**Review by TheRealMDoerr — APPROVED** [review]
LGTM.

**Pull request closed without merging**